src/reconcile/rec_write.c

```diff
@@ -1231,36 +1244,42 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 			if (WT_TXNID_LT(txnid, min_txn))
 				min_txn = txnid;
 
```

Comment:
We don't do this elsewhere and I don't think it is necessary here because there are rules around when `upd->txnid` can change that make it unnecessary.  Specifically, it is set to an active transaction ID before an update becomes visible, and can only change to `WT_TXN_ABORTED` when a transaction completes.

So, even if the code were to re-read `upd->txnid`, that is not the cause of the original problem (in fact, it would have made it less likely to fire).

The problem was that reconciliation's test for an "active transaction" for lookaside eviction was based on `txn_global->last_running`, which could change underneath reconciliation.  That could mean a transaction ID that was running concurrent with reconciliation could be cached by reconciliation and appear "committed" when the global `last_running` ID moved forward.